torchgeo/datasets/utils.py

```diff
@@ -51,9 +56,128 @@
     "draw_semantic_segmentation_masks",
     "rgb_to_mask",
     "percentile_normalization",
+    "PredictDataset",
 )
 
 
+def compute_padding(
+    original_size: Union[int, Tuple[int, int]], window_size: Union[int, Tuple[int, int]]
+) -> Tuple[int, int]:
+    """Compute required padding."""
+    original_size = cast(Tuple[int, int], _pair(original_size))
+    window_size = cast(Tuple[int, int], _pair(window_size))
+
+    if (original_size[0] % window_size[0]) == 0:
+        h_pad = 0
+    else:
+        h_pad = (window_size[0] - (original_size[0] % window_size[0])) // 2
+    if (original_size[1] % window_size[1]) == 0:
+        w_pad = 0
+    else:
+        w_pad = (window_size[1] - (original_size[1] % window_size[1])) // 2
+    return (h_pad, w_pad)
+
+
+class PredictDataset(Dataset[Any]):
```

Comment:
Why is this needed?

Reply:
I can elaborate in the prediction PR but the basic idea is to have a general PredictionDataset that represents a directory of images to run inference on.